### PR TITLE
Parse manifests

### PR DESCRIPTION
**Pull request by jyasskin**
There's no support yet for serializing the text format including manifest data, or serializing and signing the CBOR format.

### CODE REVIEW

**Comment by mrdewitt** [comment]
Does this need a rebase?  It looks like the 2 pass commit is part of this pull request.

**Comment by jyasskin** [comment]
Rebased.

File: go/webpack/signutil.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	}
+	return nil, fmt.Errorf("unknown public key type: %T", pk)
+}
+
+type rsaPSSSigner struct {
+	privKey *rsa.PrivateKey
+	hash    crypto.Hash
+}
+
+func (s rsaPSSSigner) sign(message []byte) ([]byte, error) {
```
Comment (mrdewitt) [inline go/webpack/signutil.go]: Any reason you didn't just use crypto.Signer? https://golang.org/src/crypto/crypto.go?s=3217:4212#L97

File: go/webpack/x509util.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+)
+
+// Parses a sequence of x509 certificates from filename.
+func LoadCertificatesInto(filename string, certs *[]*x509.Certificate) error {
```
Comment (mrdewitt) [inline go/webpack/x509util.go]: nit: maybe LoadCertificatesFromFile?  Into reads more like into the file to me.

Comment (jyasskin) [inline go/webpack/x509util.go]: Sure, done. I could also use just `LoadCertificates()`. I had `Into` to distinguish it from a function that I've since removed.

File: go/webpack/testdata/pki/.gitignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+*.pem
+database*
+serial*
+*.certreq
```
Comment (mrdewitt) [inline go/webpack/testdata/pki/.gitignore]: want a new line at eof?

File: go/webpack/testdata/pki/example.com.key [hunk 4]
```diff
@@ -0,0 +1,8 @@
+-----BEGIN EC PRIVATE KEY-----
```
Comment (mrdewitt) [inline go/webpack/testdata/pki/example.com.key]: Can you please add tests for each type of private key that we support?

Comment (mrdewitt) [inline go/webpack/testdata/pki/example.com.key]: Can be in another patch.

File: go/webpack/testdata/pki/.gitignore (hunk 3 above)
Comment (jyasskin) [inline go/webpack/testdata/pki/.gitignore]: Whoops, added.

File: go/webpack/testdata/pki/example.com.key (hunk 4 above)
Comment (jyasskin) [inline go/webpack/testdata/pki/example.com.key]: Yep, filed #72.

**Comment by mrdewitt** [comment]
Any reason you haven't merged this?

**Comment by jyasskin** [comment]
I was waiting for an LGTM, but I guess you approved it in your first review. I'll merge.
